content-src/styles/_mixins.scss

```diff
@@ -15,3 +15,9 @@
   max-height: 1em * $line-count * $line-height / $font-size;
   overflow: hidden;
 }
+
+@mixin dark-theme-only {
+  [lwt-newtab-brighttext]:not(.force-light-theme) & {
+    @content;
```

Comment:
Why not set the css variables to the appropriate values instead of overriding with an additional selector for each use?

https://github.com/mozilla/activity-stream/blob/master/content-src/styles/_theme.scss

Reply:
I tried both approaches and found this declarative one *far* easier to reason about and more consistent with the approach currently used for DS components.

Adding variables to every declaration that in turn point to other variables becomes incredibly cumbersome and convoluted. This approach feels far cleaner and more straightforward for maintenance.

Reply:
The variables represent a higher level concept that reflects what design is thinking of anyway. Design isn't deciding that this card should be grey-30 and that hero should also be grey-30 -- it's because dark theme text should be grey-30. With this PR, there's now two different ways to handle dark theme on the same page, but if it's preferable to maintain both, I guess that's fine.

Reply:
I'm leaning towards gvn a bit here currently.

Reply:
The css variables are used for web extension themes. See https://bugzilla.mozilla.org/show_bug.cgi?id=1347204 for more details

Reply:
I think I missed what Mardak was saying, but not sure. Maybe there are two requests here.

Are e talking about moving all of these:
```
  @include dark-theme-only {
    color: $grey-30;
  }
```

Or are we talking about updating all of these:
```
  @include dark-theme-only {
    color: $grey-30;
  }
```
to be:
```
  @include dark-theme-only {
    color: var(--newtab-text-primary-color);
  }
```

I think the way gavin has it makes sense only if this is a specific case where design is specifically asking for that element to be that colour in dark theme, or has already specifically asked for that element to not be a shared colour.

In general though, I see value in both. Gavin is probably seeing this as "I don't want to maintain ds components css in two plaecs" and Mardak is saying "We shouldn't be maintaining dark theme in two places?"

Reply:
I believe the request is to make all color values variables which are then pointed to the actually desired color variables within the theme CSS.

I see the benefit of that if there are many themes, but it feels overly abstracted and less declarative. I don't see the ROI for 2 themes and the approach I'm taking feels more consistent with how CSS is typically architected (eg: responsive layouts and overrides for specific instances being declarative and grouped).

Reply:
I hope I described this well enough, but @Mardak here is a bug to start the processes of talking about theme extensions, feel free to add comments :) https://bugzilla.mozilla.org/show_bug.cgi?id=1532353